### PR TITLE
Some more Python Interface doc cleanup

### PR DESCRIPTION
**Pull request by KurtJacobson**
See commit messages.

### CODE REVIEW

**Comment by c-morley** [comment]
The description says units per mm or degrees - but isn't it machine units (engineering units) ?

File: docs/src/config/python-interface.txt [hunk 1]
```diff
@@ -471,7 +471,7 @@ parameter, reflects [JOINT_n]FERROR.
 non-zero means max hard limit exceeded.
 
 *max_position_limit*:: '(returns float)' -
-maximum limit (soft limit) for axis motion, in machine units.configuration
+maximum limit (soft limit) for axis motion, in machine units. configuration
```
Comment (SebKuzminsky) [inline docs/src/config/python-interface.txt]: Should this say "for joint motion"?

File: docs/src/config/python-interface.txt [hunk 2]
```diff
@@ -484,7 +484,7 @@ configuration parameter, reflects [JOINT_n]MIN_FERROR.
 non-zero means min hard limit exceeded.
 
 *min_position_limit*:: '(returns float)' -
-minimum limit (soft limit) for axis motion, in machine units.configuration
+minimum limit (soft limit) for axis motion, in machine units. configuration
```
Comment (SebKuzminsky) [inline docs/src/config/python-interface.txt]: "for joint motion"?

**Comment by KurtJacobson** [comment]
@c-morley The joint units will be the same as machine units for a trivkins machines, and probably many others, but it is not guaranteed to be the same as machine units. I don't fully understand it, but according to @rene-dev on IRC "non-triv machines can have different unis, like a robot. all joints are angular, but xyz abc is mm/deg"

**Comment by c-morley** [comment]
@KurtJacobson: I agree that I am not up on non-triv at all and could be completely wrong but,
say for instance your machine is built completely in inches - It doesn't seem right to say the joints will be in mm/degree. I would think it's really joint units/degree, joint units being what ever is set in [JOINTS]. If it is actually mm then that would be the first time that I know of that linuxcnc used a specific unit internally.
But as I said this I am talking above my pay grade :) so could be absolutely missing something.

**Comment by KurtJacobson** [comment]
@c-morley Now I understand your concern, I did not read your first comment properly! LinuxCNC does indeed use millimeter internally, and all other units are converted to millimeter. So, strange as it seems, the value of [JOINT]UNITS is the number of machine units that fit into one millimeter. For an inch machine the units value would be 0.03937in/mm, for a metric machine it would be 1mm/mm. For rotary axes the internal units is deg, so if you configure an axis in grad the units value would be 1.1111grad/deg etc. It took me quite some time to figure out was was going on, and that is what prompted me to try and clarify it in the docs.

**Comment by c-morley** [comment]
@KurtJacobson  just revisiting your comments.
I wonder if we should add your description of how to calculate joint units - I find it non-intuitive.?

**Comment by KurtJacobson** [comment]
@c-morley it probably would be good to include a description of how the joint units are calculated, but I doubt anybody even used them, so it may not be worth the effort.

**Comment by KurtJacobson** [comment]
Thank you @c-morley!

**Pull request merged**